x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupIndexerAction.java

```diff
@@ -186,5 +198,32 @@ protected void finishHim() {
                 listener.onFailure(e);
             }
         }
+
+        @Override
+        protected void onOperation(@Nullable ShardRouting shard, final ShardIterator shardIt, int shardIndex, Exception e) {
+            // when this shard operation failed, cancel other shard operations
+            cancelOtherShardIndexers();
+            super.onOperation(shard, shardIt, shardIndex, e);
+        }
+
+        private void cancelOtherShardIndexers() {
+            if (false == hasCancelled) {
+                cancelTasksAction.execute(
+                    task,
+                    new CancelTasksRequest().setTargetParentTaskId(new TaskId(clusterService.localNode().getId(), task.getId())),
+                    ActionListener.wrap(r -> {
+                        logger.info("[{}] rollup cancel other shard indexers", request.getRollupRequest().getSourceIndex());
```

Comment:
Maybe check the `r` response here? To log whether tasks are really cancelled?

Reply:
done

Reply:
I also don't see this change. Maybe you forgot to push it?